corehq/apps/reports/standard/sms.py

```diff
@@ -170,7 +167,18 @@ def _fmt_contact_link(self, recipient_id, doc_info, extra_text=None):
         username = username or "-"
         contact_type = contact_type or _("Unknown")
         if url:
-            ret = self.table_cell(username, '<a target="_blank" href="%s">%s</a>' % (url, username))
+            # We need to store the contact type and the recipient ID in the contact link element, because
+            # MessageLogReport can only use the link, not the whole table cell dictionary, and so it won't have
+            # table_cell['raw'] available for export. export_table will read contacttype and recipientid
+            # attributes to populate those columns in the export spreadsheet.
+            ret = self.table_cell(
+                username,
+                '<a target="_blank" href="{url}" contacttype="{type}" recipientid="{id}">{username}</a>'.format(
+                    username=username,
+                    url=url,
+                    type=contact_type,
+                    id=recipient_id or ''
+                ))
```

Comment:
I'm not sure about this approach. If you look at `GenericTabularReport.export_table` you'll see that it calls `GenericTabularReport.export_rows` which you can override and add in additional formatting etc. In `export_rows` you do have access to `cell['raw']`